File: Mixter/Domain/MessageId.cs
using System;

namespace Mixter.Domain
{
    public struct MessageId
    {
        private MessageId(string id) 
            : this()
        {
            Id = id;
        }

        public string Id { get; private set; }

        public static MessageId Generate()
        {
            return new MessageId(Guid.NewGuid().ToString());
        }

        public override string ToString()
        {
            return Id;
        }
    }
}

File: src/main/java/mixter/domain/message/MessageId.java
package mixter.domain.message;

import java.util.UUID;

class MessageId {
    private final String id;

    public MessageId() {
        this.id = UUID.randomUUID().toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        MessageId messageId = (MessageId) o;

        return id.equals(messageId.id);

    }

    @Override
    public int hashCode() {
        return id.hashCode();
    }

    @Override
    public String toString() {
        return id;
    }

    public String getId() {
        return id;
    }
}
